### ISSUE TITLE
Report congestion control algorithm

### ISSUE THREAD

**Issue by bmah888** [esnet/iperf#461]
For some upcoming ESnet experiments, it's useful to put the congestion control algorithm being used for a test in the test's output.  (I'm presuming that just putting it in the JSON is fine for our use case.)

**Comment by bmah888** [comment]
Note:  Do this for TCP tests on systems that support setting the congestion control algorithm (Linux, FreeBSD).  
```
val = char[TCP_CA_NAME_MAX];
len = TCP_CA_NAME_MAX;
getsockopt(s, IPPROTO_TCP, TCP_CONGESTION, val, &len);
```
Need to do the above on the sending side of the test (which might not be the client!).  client can emit this in JSON.

**Comment by bmah888** [comment]
There's something that doesn't seem right about the way we set the congestion control algorithm...it's tied too closely to the low-level TCP operations of connecting and accepting, and probably needs to be pulled up into the iperf_client_api.c and iperf_server_api.c files where the congestion control algorithms can be set only for those TCP connections that are used for data streams.

**Comment by bmah888** [comment]
@bltierney:  Have you gotten a chance to try this yet?  If this is what you had in mind I can merge it to master and the 3.1 branch.

**Comment by bltierney** [comment]
sorry for the delay; tested and working.

**Comment by bmah888** [comment]
Merged accumulated commits to 3.1-STABLE.
